http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesServiceDbQueries.scala

```diff
@@ -60,8 +61,39 @@ object KubernetesServiceDbQueries {
     )
 
   /**
-   * List all apps that need monitoring. Called by MonitorAtBoot.
-   */
+    * List all RUNNING apps that are not on the given target chart version. Used to decide which apps to update.
+    */
+  def listAppsForUpdate(targetVersion: Chart,
+                        appType: AppType,
+                        cloudProvider: CloudProvider,
+                        chartVersionsToInclude: List[Chart] = List(),
+                        chartVersionsToExclude: List[Chart] = List(),
+                        googleProject: Option[GoogleProject] = None,
+                        workspaceId: Option[WorkspaceId] = None,
+                        appNames: List[AppName] = List()
+  )(implicit
+    ec: ExecutionContext
+  ): DBIO[List[KubernetesCluster]] =
+    joinFullAppAndUnmarshal(
+      kubernetesClusterQuery
+        .filter(_.cloudProvider === cloudProvider)
+        .filterOpt(googleProject) { case (clusterTable, gp) =>
+          clusterTable.cloudContextDb === CloudContextDb(gp.value)
+        },
+      nodepoolQuery,
+      appQuery
+        .filter(_.status === (AppStatus.Running: AppStatus))
+        .filter(_.appType === appType)
+        .filter(_.chart =!= targetVersion)
+        .filterIf(chartVersionsToInclude.nonEmpty)(_.chart inSetBind chartVersionsToInclude)
```

Comment:
Fancy slick stuff :) Looks good to me, may be a good candidate to unit test (given different input, does the query return the correct things..)

Reply:
Oh indeed, we tested all possible combination on the BEE using the dryRun mode, but I did not think of transposing this to unit tests. The tricky part is having the correct older versions of the apps installed, but that might be mockable